dotnet/src/Connectors/Connectors.OpenAI/AzureSdk/ClientCore.cs

```diff
@@ -1544,4 +1513,154 @@ private void LogUsage(CompletionsUsage usage)
             await functionCallCallback(context).ConfigureAwait(false);
         }
     }
+
+    /// <summary>
+    /// Configures the function calling functionality based on the provided parameters.
+    /// </summary>
+    /// <param name="kernel">The <see cref="Kernel"/> to be used for function calling.</param>
+    /// <param name="executionSettings">Execution settings for the completion API.</param>
+    /// <param name="chatOptions">The chat completion options from the Azure.AI.OpenAI package.</param>
+    /// <param name="requestIndex">Request sequence index of automatic function invocation process.</param>
+    private (bool? AllowAnyRequestedKernelFunction, int? MaximumAutoInvokeAttempts)? ConfigureFunctionCalling(Kernel? kernel, OpenAIPromptExecutionSettings executionSettings, ChatCompletionsOptions chatOptions, int requestIndex)
+    {
+        (bool? AllowAnyRequestedKernelFunction, int? MaximumAutoInvokeAttempts)? result = null;
+
+        // If neither behavior specified, we don't need to do anything.
+        if (executionSettings.FunctionChoiceBehavior is null && executionSettings.ToolCallBehavior is null)
+        {
+            return result;
+        }
+
+        // If both behaviors are specified, we can't handle that.
+        if (executionSettings.FunctionChoiceBehavior is not null && executionSettings.ToolCallBehavior is not null)
+        {
+            throw new ArgumentException($"{nameof(executionSettings.ToolCallBehavior)} and {nameof(executionSettings.FunctionChoiceBehavior)} cannot be used together.");
+        }
+
+        // Set the tool choice to none. If we end up wanting to use tools, we'll set it to the desired value.
+        chatOptions.ToolChoice = ChatCompletionsToolChoice.None;
+        chatOptions.Tools.Clear();
+
+        // Handling new tool behavior represented by `PromptExecutionSettings.FunctionChoiceBehavior` property.
+        if (executionSettings.FunctionChoiceBehavior is { } functionChoiceBehavior)
+        {
+            result = this.ConfigureFunctionCalling(kernel, chatOptions, requestIndex, functionChoiceBehavior);
+        }
+        // Handling old-style tool call behavior represented by `OpenAIPromptExecutionSettings.ToolCallBehavior` property.
+        else if (executionSettings.ToolCallBehavior is { } toolCallBehavior)
+        {
+            result = this.ConfigureFunctionCalling(kernel, chatOptions, requestIndex, toolCallBehavior);
+        }
+
+        // Having already sent tools and with tool call information in history, the service can become unhappy ("[] is too short - 'tools'")
+        // if we don't send any tools in subsequent requests, even if we say not to use any.
+        // Similarly, if we say not to use any tool (ToolChoice = ChatCompletionsToolChoice.None) and dont provide any for the first request,
+        // the service fails with "'tool_choice' is only allowed when 'tools' are specified."
```

Comment:
Yes, it is still the case; the service fails with the error:
<img width="850" alt="image" src="https://github.com/microsoft/semantic-kernel/assets/68852919/35b939af-440c-467a-8d80-bb74c7472488">
I've tried to assign null to the `ToolChoice` property because it's null by default, even though it's not nullable, but it hasn't changed anything.